q2_composition/_ancombc.py

```diff
@@ -72,11 +73,33 @@ def _leaf_collector(term):
     return _leaf_collector(term[1]) + _leaf_collector(term[2])
 
 
+def _get_ref_level_defaults_from_formula_terms(metadata, term,
+                                               reference_levels):
+    term_alpha_value = (metadata.get_column(term)
+                        .to_dataframe()
+                        .sort_values(term)[term][0])
+    ref_level_pair = term + '::' + str(term_alpha_value)
+    reference_levels.append(ref_level_pair)
+
+    return reference_levels
+
+
 def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
              reference_levels, neg_lb, tol, max_iter, conserve, alpha):
 
     meta = metadata.to_dataframe()
 
+    md_column_types = {}
+    for name, attrs in metadata.columns.items():
+        if attrs[0] == 'numeric':
+            md_column_types[name] = 'numeric'
+        elif attrs[0] == 'categorical':
+            md_column_types[name] = 'categorical'
+        else:
+            pass
+
+    md_column_types_json = json.dumps(md_column_types)
```

Comment:
I'm assuming from this that `attrs[0]` could be the type of the column in which case we want to set the type in the dict, or it could be something else entirely in which case we ignore it, so I am assuming there are valid values `attrs[0]` could have that are not the column type? Not that that's necessarily a problem, just trying to understand because I would normally assume that `attrs[0]` would always be referring to the same thing (in this case the column type).

Reply:
Yeah good question @Oddant1 - this refers to the metadata column type, which will always be either numeric or categorical. This shouldn't ever end up being anything else, but I decided to add that third branch just to be safe. With that being said, I talked w/Evan about this and it's probably better to have this third branch error in case we ever add any other metadata column types in the future.

Reply:
Yeah, that's what I was going to say.